flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/mailbox/Mailbox.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.runtime.tasks.mailbox;
+
+import javax.annotation.Nonnull;
+
+/**
+ * A mailbox is basically a blocking queue for inter-thread message exchange in form of {@link Runnable} objects between
+ * multiple producer threads and a single consumer.
+ */
+public interface Mailbox extends MailboxReceiver, MailboxSender {
+
+	/**
+	 * The effect of this is that all pending letters are dropped and the given priorityAction
+	 * is enqueued to the head of the mailbox.
+	 *
+	 * @param priorityAction action to enqueue atomically after the mailbox was cleared.
+	 */
+	void clearAndPut(@Nonnull Runnable priorityAction);
```

Comment:
I think `@Nonnull` is redundant and we should implicitly assume that parameters are not nullable unless stated differently.

Reply:
There is an additional benefit from adding this, which is that the compiler introduces an assertion for `!= null` if you run the code with `-ea`.

Reply:
Hmmm, how useful is that? Shouldn't we use `checkState`/`checkArgument`/`checkNotNull`/`requireNonNull` anyway? If not, shouldn't we add this to our code style?

Reply:
Hmm, I would suggest to do it so because it has 3 benefits over the other options:
- Shorter to write and part of the signature makes it more visible
- IDE can show warnings in many cases when you try to violate it.
- Can be de/activated in production code.

Reply:
Ok. I think it would be better to have some one common policy in this regard for the future, so maybe we will need to follow up on this at some point.